juju/model.py

```diff
@@ -31,6 +31,22 @@ def relation(self, relation_name):
             # ideally integrating the error catching with Juju's mechanisms.
             raise TooManyRelatedApps(relation_name, num_related, 1)
 
+    def relation_by_id(self, relation_name, relation_id):
+        def _norm(relation_id):
+            # Relation IDs can be in the form '{relation_name}:{id}', so we need to
+            # normalize them to just '{id}' for comparison.
+            return relation_id.split(':')[-1]
+
+        for relation in self.relations[relation_name]:
+            if _norm(relation.relation_id) == _norm(relation_id):
```

Comment:
Relevance depends on the outcome of the prior comment, but if this logic stays we shouldn't be normalizing the exact same data every iteration of the loop.